src/main/java/net/dv8tion/jda/api/entities/Role.java

```diff
@@ -373,5 +373,47 @@ default String getIntegrationId()
         {
             return isIntegration() ? Long.toUnsignedString(getIntegrationIdLong()) : null;
         }
+
+        /**
+         * Whether this role can be acquired through a premium subscription purchase.
+         *
+         * @return True, if this is a subscription role
+         *
+         * @see    #getSubscriptionIdLong()
+         */
+        boolean hasSubscriptionListing();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or 0 if this role is not for a subscription listing
+         */
+        long getSubscriptionIdLong();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or null if this role is not for a subscription listing
+         */
+        @Nullable
+        default String getSubscriptionId()
+        {
+            return hasSubscriptionListing() ? Long.toUnsignedString(getSubscriptionIdLong()) : null;
+        }
+
+        /**
+         * Whether this role can be purchased by a user.
+         *
+         * @return True, if this role is purchasable
+         */
+        boolean isAvailableForPurchase();
+
+        /**
+         * Whether this role is acquired through a user connection.
+         * <br>Such as twitter or reddit.
```

Comment:
I think we should call out that this can (and does) represent roles granted via 3rd party applications too, not just connections that feel like 1st party integrations.